src/idpi/operators/support_gis.py

```diff
@@ -0,0 +1,229 @@
+    requirements = {
+        "angleOfRotationInDegrees": 0.0,
+        "gridType": "rotated_ll",
+        "iScansNegatively": 0,
+        "jScansPositively": 1,
+    }
+    errors = {key: geo[key] for key in requirements if requirements[key] != geo[key]}
+    if errors:
+        msg = f"Unsupported values for keys: {errors}"
+        raise RuntimeError(msg)
+
+
+def get_grid(geo: Mapping[str, typing.Any]) -> RotLatLonGrid:
+    """Get grid parameters for a given field.
+
+    Parameters
+    ----------
+    geo : Mapping[str, Any]
+        Grib keys related to the geography of the field.
+
+    Raises
+    ------
+    ValueError
+        if the field is not defined on a rotated lat lon grid.
+
+    Returns
+    -------
+    RotLatLonGrid
+        object representing the rotated lat lon grid.
+
+    """
+    _check_requirements(geo)
+
+    ni = geo["Ni"]
+    x0 = geo["longitudeOfFirstGridPointInDegrees"]
+    dx = geo["iDirectionIncrementInDegrees"]
+    rlon = get_grid_coords(ni, x0, dx, "x")
+
+    nj = geo["Nj"]
+    y0 = geo["latitudeOfFirstGridPointInDegrees"]
+    dy = geo["jDirectionIncrementInDegrees"]
+    rlat = get_grid_coords(nj, y0, dy, "y")
+
+    lon_np = (geo["longitudeOfSouthernPoleInDegrees"] - 180) % 360
+    lat_np = -1 * geo["latitudeOfSouthernPoleInDegrees"]
+
+    return RotLatLonGrid(rlon, rlat, lon_np, lat_np)
+
+
+def rot2geolatlon(grid: RotLatLonGrid) -> tuple[xr.DataArray, xr.DataArray]:
+    """Compute geographical lat lon values for a rotated lat lon grid.
+
+    Parameters
+    ----------
+    grid : RotLatLonGrid
+        object representing the rotated lat lon grid.
+
+    Returns
+    -------
+    tuple[xarray.DataArray, xarray.DataArray]
+        tuple of longitudes and latitudes of the grid points in the
+        geographical lat lon CRS.
+
+    """
+    deg2rad = np.pi / 180
+    rad2deg = 180 / np.pi
+    sin_np_lat = np.sin(deg2rad * grid.north_pole_lat)
+    cos_np_lat = np.cos(deg2rad * grid.north_pole_lat)
+
+    sin_np_lon = np.sin(deg2rad * grid.north_pole_lon)
+    cos_np_lon = np.cos(deg2rad * grid.north_pole_lon)
+
+    # Compute new coordinates
+    # ... normalize input coordinates
+    norm_lon = deg2rad * grid.rlon.where(grid.rlon < 180, grid.rlon - 360)
+
+    # ... cache trigonometric operations
+    sin_lat = np.sin(deg2rad * grid.rlat)
+    cos_lat = np.cos(deg2rad * grid.rlat)
+    sin_lon = np.sin(norm_lon)
+    cos_lon = np.cos(norm_lon)
+    # ... compute latitude
+    arg1 = cos_np_lat * cos_lat * cos_lon + sin_np_lat * sin_lat
+    lat = rad2deg * np.arcsin(arg1)
+    # ... compute longitude
+    arg1 = (
+        sin_np_lon * (-sin_np_lat * cos_lon * cos_lat + cos_np_lat * sin_lat)
+        - cos_np_lon * sin_lon * cos_lat
+    )
+    arg2 = (
+        cos_np_lon * (-sin_np_lat * cos_lon * cos_lat + cos_np_lat * sin_lat)
+        + sin_np_lon * sin_lon * cos_lat
+    )
+    arg2 = xr.where(np.abs(arg2) < 1e-20, 1e-20, arg2)  # BUG: changes sign?
```

Comment:
arg2 changes sign always or only when 1e-20? Think this comment should be expanded slightly.

Reply:
When arg2 is negative but the absolute value is less than 1e-20, it is replaced by a value that is positive.

Reply:
Ah ok changes sign but not == 1e-20 as expected?

Reply:
I expanded the comment. Does it make more sense now?

Reply:
Sorry my question was whether the value/modulus changes as well as the sign or not.

Reply:
Because the expected sign is positive, expected value is 1e-20 ? Or do i misunderstand?

Reply:
I think it's just a guard to avoid a division by zero error. Arctan of +inf is 0.5 pi and -inf is -0.5 pi. The modulus does not catch the sign change. This probably went undetected because the range of values for which it misbehaves is tiny.

Reply:
I understand the need, I was asking about the value and not the sign, but nevermind.